### ISSUE TITLE
sys_mmap failed rax: 0xfffffffffffffff4

### ISSUE THREAD

**Issue by jjoshua2** [tthsqe12/asm#18]
Bench runs fine, but after analyzing a few positions with go depth 31, I get this error. I was using 16 cores with 24GB of memory.

**Comment by tthsqe12** [comment]
is you negate rax, you get 12, and this is
#define ENOMEM          12      /* Out of memory */

**Comment by jjoshua2** [comment]
Ok, I just requested 8 cores with 10gb of memory and am running a bench 8196 8 31 and I will see if if that works, there is a chance I was using too much memory and the process was killed? Or is that a memory allocation failed?

**Comment by tthsqe12** [comment]
there should be no memory allocation while searching in bench, ect..
the exception is if tablebases are being used, in which case mmap is called sometimes

**Comment by jjoshua2** [comment]
Ok, that is probably it, I am using 6 piece tablebases. The bench finished without errors, so I ran my analysis with 16 cores and 23gb limit on memory, while the hash table is only 16gb and finished analyzing 11 positions to depth 31 before it  got sys_mmap failed rax: 0xfffffffffffffff4.

**Comment by jjoshua2** [comment]
With 8gb hashtable it analyzed 14 positions before it ran out of memory. Must be some trouble with loading the tablebases into memory and not clearing them?

**Comment by tthsqe12** [comment]
of course. The six man table bases are huge and should quickly hit your 23gb limit.

**Comment by jjoshua2** [comment]
But I use 6 pieces on my quad core with only 16gb of memory on my Windows box. How can I get it to not use more memory than it has and crash?

**Comment by jjoshua2** [comment]
I got 64000mb of memory now, and I have half that again as swap, so it can use 96gb now. How can I see how much memory the tablebases are using?

**Comment by jjoshua2** [comment]
I believe that with a 16gb hashtable and 6 piece TB it needs 84300160 kb of virtual memory, which is roughly the 16gb hash +less than 70gb bases. I've been running it for a while with 96gb of memory and seems to have maxed out without crashing.

**Comment by jjoshua2** [comment]
I think this issue can be closed now, since it works if you have enough virtual memory. It does seem that the program never deallocates the virtual memory when it isn't accessing a TB, so over time it grows to be the size of all 6 piece TB, but that probably makes it faster on systems that don't have limited amounts of virtual memory.
